stan/math/prim/mat/meta/broadcast_array.hpp

```diff
@@ -0,0 +1,29 @@
+#include <stan/math/prim/scal/meta/broadcast_array.hpp>
+#include <Eigen/Dense>
+#include <stdexcept>
+
+#ifndef STAN_MATH_PRIM_MAT_META_BROADCAST_ARRAY_HPP
+#define STAN_MATH_PRIM_MAT_META_BROADCAST_ARRAY_HPP
+
+namespace stan {
+  namespace math {
+    namespace internal {
+      template <typename ViewElt, typename OpElt, int R, int C>
+      class empty_broadcast_array<ViewElt, Eigen::Matrix<OpElt, R, C>> {
+      public:
+        empty_broadcast_array() {}
+        ViewElt& operator[] (int /*i*/) {
+          throw std::logic_error("Don't do this");
```

Comment:
"Don't do this" is *not* a sufficient error message.  Maybe something like "illegal call to operator[](int), ..." where the ... is the explanation of why this error's getting called.

Reply:
This is my fault; he's just copying my original `empty_broadcast_array` semantics

Reply:
Wherever it came from, it needs to be fixed.

Reply:
Is there a reason to do this kind of error message for something like this (that will be caught at development time) when the exception will come with a traceback?

Reply:
Why are there functions implemented to throw exceptions?  I had assumed this was a base class that needed to compile and link.

Reply:
Interesting; apparently C++ doesn't actually require an implementation for these since they are never actually called? Should we just remove the implementations (and perhaps add a comment)?

Reply:
Yes.  Alternatively, give them `= 0` definitions.

Reply:
OK. I removed the implementations and added a comment.